src/components/WalletModal/index.tsx

```diff
@@ -244,28 +209,48 @@ export default function WalletModal({
       const isReconnect =
         connectedWallets.filter((wallet) => wallet.account === account && wallet.walletType === walletType).length > 0
       sendAnalyticsEventAndUserInfo(account, walletType, chainId, isReconnect)
-      setShouldLogWalletBalances(true)
+      setShouldLogNativeBalance(true)
+      setShouldLogUsdcBalance(true)
+      setShouldLogWethBalance(true)
       if (!isReconnect) addWalletToConnectedWallets({ account, walletType })
     }
     setLastActiveWalletAddress(account)
   }, [connectedWallets, addWalletToConnectedWallets, lastActiveWalletAddress, account, connector, chainId])
 
-  // Send wallet balance info once it becomes available.
+  // Send wallet balances info once it becomes available.
   useEffect(() => {
-    if (!tokenBalancesIsLoading && shouldLogWalletBalances && balances && nativeCurrencyBalanceUsdValue) {
+    if (shouldLogNativeBalance && nativeCurrencyBalanceUsdValue && nativeCurrencyBalance) {
       const nativeCurrencyBalanceUsd =
         native && nativeCurrencyBalanceUsdValue ? parseFloat(nativeCurrencyBalanceUsdValue) : 0
-      sendAnalyticsWalletBalanceUserInfo(balances, nativeCurrencyBalanceUsd)
-      setShouldLogWalletBalances(false)
+      const nativeCurrencyBalanceAmount = formatToDecimal(
+        nativeCurrencyBalance,
+        nativeCurrencyBalance.currency.decimals
+      )
+      user.set(CUSTOM_USER_PROPERTIES.WALLET_NATIVE_CURRENCY_AMOUNT, nativeCurrencyBalanceAmount)
+      user.set(CUSTOM_USER_PROPERTIES.WALLET_NATIVE_CURRENCY_BALANCE_USD, nativeCurrencyBalanceUsd)
+      setShouldLogNativeBalance(false)
+    }
+  }, [nativeCurrencyBalanceUsdValue, nativeCurrencyBalance, shouldLogNativeBalance, setShouldLogNativeBalance, native])
+
+  useEffect(() => {
```

Comment:
can you generalize this effect to use? very related logic between the three effects here

Reply:
extracted to custom hook as per offline guidance, hope this looks good?